### ISSUE TITLE
Flutter Doctor crashes on "Connected device" with message "Exception: ideviceinfo returned an error: ERROR: Could not connect to lockdownd, error code -8"

### ISSUE THREAD

**Issue by martintjandra** [flutter/flutter#49196]
## Steps to Reproduce

Almost all of these steps producing the same result: **Exception: ideviceinfo returned an error: ERROR: Could not connect to lockdownd, error code -8**

Test 1
1. Setup new Flutter project.
2. Run Flutter Doctor 
3. Error.

Test 2
1. Setup new Flutter project.
2. Open from Android Studio
3. Run Flutter Doctor from Android Studio
4. Error.

Test 3 
1. Setup new Flutter project.
2. Open from Android Studio
3. Open Android Simulator
4. Run Flutter Doctor from Android Studio
5. Error.

Test 4
1. Open existing Flutter project that once can run correctly.
2. Compile and run.
3. Error.

Test 5
1. Open existing Flutter project that once can run correctly.
2. Select Android simulator to install from dropdown.
3. Select Open Android emulator: Pixel 2 API 28 Pie.
4. Android emulator open.
5. Compile and run.
6. Error.

Test 6
1. Open existing Flutter project that once can run correctly.
2. Select device to install from dropdown.
3. Select Open Android emulator: Pixel 2 API 28 Pie.
4. Android emulator open.
5. Reselect device to install from dropdown, the android emulator isn't there.
6. Reselect Open Android emulator.
7. Error: Error Opening Emulator. 
emulator: WARNING: Crash service did not start.
emulator: ERROR: Running multiple emulators with the same AVD is an experimental feature.

Test 7
1. Downloading the latest Flutter SDK.
2. Updating Android Studio, Android SDK, and Emulator to the latest.
3. Open Project Structure
4. Open "Project" menu under Project Settings.
5. Select the latest Android SDK.
6. Open "Module" menu under Project Settings.
7. Select the latest Android SDK for each modules.
8. Open Android Emulator.
9. Compile and Run.
10. Still the same error.

I have checked:
- https://github.com/flutter/flutter/issues/35035
- https://github.com/flutter/flutter/issues/37236 
- https://github.com/flutter/flutter/issues/36789

All of these are about iOS device not trusting the computer. This is not about iOS device, but Android emulator. I used to be able to compile to my Android emulators correctly without any issue. Now suddenly, it doesn't work.

## Logs

```
/Volumes/Data/Frameworks/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.13.6 17G9016, locale en-US)
    • Flutter version 1.12.13+hotfix.5 at /Volumes/Data/Frameworks/flutter
    • Framework revision 27321ebbad (6 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Volumes/Data/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Volumes/Data/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ Flutter requires a minimum Xcode version of 10.2.0.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.40.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.6.0

[☠] Connected device (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at https://github.com/flutter/flutter/issues.
    ✗ Exception: ideviceinfo returned an error:
      ERROR: Could not connect to lockdownd, error code -8

    • #0      IMobileDevice.getInfoForDevice (package:flutter_tools/src/ios/mac.dart:226:9)
      <asynchronous suspension>
      #1      IOSDevice.getAttachedDevices (package:flutter_tools/src/ios/devices.dart:190:55)
      <asynchronous suspension>
      #2      IOSDevices.pollingGetDevices (package:flutter_tools/src/ios/devices.dart:115:57)
      #3      PollingDeviceDiscovery.devices (package:flutter_tools/src/device.dart:285:52)
      #4      DeviceManager.getAllConnectedDevices (package:flutter_tools/src/device.dart:140:46)
      <asynchronous suspension>
      #5      DeviceValidator.validate (package:flutter_tools/src/doctor.dart:851:54)
      #6      asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:32)
      #7      _rootRun (dart:async/zone.dart:1126:13)
      #8      _CustomZone.run (dart:async/zone.dart:1023:19)
      #9      _runZoned (dart:async/zone.dart:1518:10)
      #10     runZoned (dart:async/zone.dart:1502:12)
      #11     asyncGuard (package:flutter_tools/src/base/async_guard.dart:109:3)
      #12     Doctor.startValidatorTasks (package:flutter_tools/src/doctor.dart:158:9)
      #13     Doctor.diagnose (package:flutter_tools/src/doctor.dart:252:41)
      #14     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:45:6)
      #15     Doctor.diagnose (package:flutter_tools/src/doctor.dart:241:24)
      #16     DoctorCommand.runCommand (package:flutter_tools/src/commands/doctor.dart:59:39)
      #17     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:45:6)
      #18     DoctorCommand.runCommand (package:flutter_tools/src/commands/doctor.dart:45:42)
      #19     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:615:18)
      #20     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
      #21     _rootRunUnary (dart:async/zone.dart:1134:38)
      #22     _CustomZone.runUnary (dart:async/zone.dart:1031:19)
      #23     _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
      #24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
      #25     Future._propagateToListeners (dart:async/future_impl.dart:709:32)
      #26     Future._completeWithValue (dart:async/future_impl.dart:524:5)
      #27     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:554:7)
      #28     _rootRun (dart:async/zone.dart:1126:13)
      #29     _CustomZone.run (dart:async/zone.dart:1023:19)
      #30     _CustomZone.runGuarded (dart:async/zone.dart:925:7)
      #31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:965:23)
      #32     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
      #33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
      #34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
      #35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:175:5)


! Doctor found issues in 2 categories.
Process finished with exit code 0
```

**Comment by jmagman** [comment]
So you have no iOS device plugged in and aren't running Instruments (iOS simulator)?

**Comment by martintjandra** [comment]
Sorry, found the culprit. Yes, this is actually because of there is iOS device plugged in. I got curious. So when I tried to open iTunes to access the device, the iTunes cannot access the device (like [this](http://khellek.blogspot.com/2012/10/itunes-could-not-connect-to-this-iphone.html)). I can't get the device to show "Trust this computer" message as well. I have tried to remove, recreate, and set correct permission for /var/db/lockdown (like [this](https://stackoverflow.com/a/28342269/3003927)), but when I reconnect the device, the result is still the same. But I can charge the device using the cable. So I guess this must be because somehow the cable is faulty or something.

Once I unplug the iOS device, and "restart" the daemon using the change project SDK steps, the Android Emulators/Devices show up correctly.

This still means that I can't compile to my iOS device though. But I guess that's for another day. Thanks for the help!
